python/ray/tune/schedulers/pbt.py

```diff
@@ -231,8 +231,16 @@ def on_trial_add(self, trial_runner, trial):
         self._trial_state[trial] = PBTTrialState(trial)
 
     def on_trial_result(self, trial_runner, trial, result):
-        if self._time_attr not in result or self._metric not in result:
-            return TrialScheduler.CONTINUE
+        if self._time_attr not in result:
+            raise TuneError("Cannot find time_attr {} in trial results. "
```

Comment:
Actually I just realized we shouldn't do this - the reason is because the user can return a result in `tune.report` that doesn't necessarily have the optimization metric.

What you should do instead is just log somewhat frequently until the metric is seen (you can use something like ray.util.enable_periodic_logging), but also provide an option for the user to turn off the error message

Reply:
in what situations would the user want to do that? If the metric is not found in the result then PBT doesnt actually do anything iirc

Reply:
one common example is training vs validation metrics. you may want to log a bunch of training metrics which occur at a high frequency, but optimize based on validation loss which occurs at a lower frequency.